plugins/modules/vmware_guest.py

```diff
@@ -2218,7 +2219,10 @@ def customize_vm(self, vm_obj):
 
             ident.identification = vim.vm.customization.Identification()
 
-            if self.params['customization'].get('password', '') != '':
+            if self.params['customization'].get('password', '') is None:
```

Comment:
Will this ever be true? If `params['customization']['password']` isn't set, it returns an empty string `''` which is _not_ `None`.

```
>>> params={}
>>> params['customization']={}
>>> params['customization'].get('password', '') is None
False
>>>
```

Reply:
OK, you got the point, I didn't check what the `get` function really does and it would make sense if it worked like that in the original code. But for me, it doesn't work the way you demonstrated. The password field does exist and is set to `None` and that's why the condition worked for me when testing.

I tried to insert the following code:
```
self.module.warn(str(self.params['customization']))
```
When running `ansible-playbook -i testvm, playbook.yaml` against this playbook:
```yaml
- name: Customize existing VM
  hosts: all
  gather_facts: false
  tasks:
  - delegate_to: localhost 
    community.vmware.vmware_guest:
      hostname: vcenter
      username: administrator@vsphere.local
      password: password
      datacenter: datacenter
      datastore: datastore
      name: '{{ inventory_hostname }}'
      validate_certs: no
      state: poweredon
      template: Win2019
      networks:
      - name: port-group
      customization:
        existing_vm: true
        hostname: '{{ inventory_hostname }}'
```
I can see this output for the inserted code:
```
[WARNING]: {'existing_vm': True, 'hostname': 'testvm', 'autologon': None, 'autologoncount': None, 'dns_servers': None, 'dns_suffix': None, 'domain': None, 'domainadmin': None, 'domainadminpassword': None, 'fullname': None, 'hwclockUTC': None, 'joindomain':
None, 'joinworkgroup': None, 'orgname': None, 'password': None, 'productid': None, 'runonce': None, 'timezone': None}
```
So all the fields, that are not passed to the module are indeed present with a value of `None`. This also explains why some other conditions testing `None` work (like `timezone` or `autologon`). Or what am I missing?

Reply:
Thanks, I updated the condition.

Regarding whether the field will be mandatory - the Administrator password was emptied only in case the password value was set to `''`. If the value was not set, the user password was reset to 'None', literally. I don't think that is expected, at least it is undocumented.
Is it ok to keep the password field mandatory or should we choose a different approach?